src/experiments/preferences/api.js

```diff
@@ -47,40 +47,34 @@ ExtensionPreferencesManager.addSetting("dohRollout.state", {
   },
 });
 
-const prefManager = {
-  prefHasUserValue(name) {
-    return Services.prefs.prefHasUserValue(name);
-  },
-  getUserPref(name, value) {
-    if (!Services.prefs.prefHasUserValue(name)) {
-      return value;
-    }
-    let type = Services.prefs.getPrefType(name);
-    switch (type) {
-    case Services.prefs.PREF_STRING:
-      return Services.prefs.getCharPref(name, value);
-    case Services.prefs.PREF_INT:
-      return Services.prefs.getIntPref(name, value);
-    case Services.prefs.PREF_BOOL:
-      return Services.prefs.getBoolPref(name, value);
-    default:
-      throw new Error("Unknown type");
-    }
-  }
-};
+function getBoolPref(name, defaultValue=null) {
```

Comment:
I'd suggest leaving these as null. If the default value is non-null, it isn't possible to tell the difference between getting the default back (because no value was set) and getting back a value that was set that happened to match the default value.

Reply:
I realize now that my previous comment is wrong, since this just gets forwarded to `get*Pref`, which insists on having the same type for the default value.